### ISSUE TITLE
some no proper `no visible binding` by `object_usage_linter`

### ISSUE THREAD

**Issue by iago-pssjd** [r-lib/lintr#787]
For the next code:
```r
library(dplyr)
library(magrittr)
test <- function(data) {
  f <<- function(y) class
  data %T>% {
      print(summary(.))
  } %>%
    head()
}
```
I get
```r
lintr::lint('file.r')
/home/iago/Downloads/rep.r:7:3: warning: no visible binding for '<<-' assignment to ‘f’
  f <<- function(y) class
  ^
/home/iago/Downloads/rep.r:9:21: warning: no visible binding for global variable ‘.’
      print(summary(.))
```
So it seems to not understand global assignments of anonymous functions inside functions (but if the assignment is previously done outside the `test` function, there is no warning), and also the `.` used in `magrittr` pipe workflow.

Further, usually I write such code as follows:
```r
library(dplyr)
library(magrittr)
test <- function(data) {
  f <<- function(y) class
  data %T>% 
     {print(summary(.))} %>%
     head()
}
```
But then I get a lot of style messages due to `{`, `}` positions...

Thank you!

**Comment by AshesITR** [comment]
Hello @iago-pssjd.
Thanks for taking the time to report an issue. I will try to answer a few of your findings:

1. The warnings you get in your first example are from `codetools::checkUsage()`.  
Using `<<-` is meant for actively overwriting an _existing_ binding in a parent environment. Since the binding `f` doesn't exist, you receive a warning.
To better show your intent with the same effect, use `assign("f", ..., envir = parent.frame())`. This will always assign in the immediate parent frame whereas `<<-` could overwrite in a different frame if `f` is added as a binding somewhere.
2. The `.` is a special syntax used by magrittr. If you want to suppress this message, you can use `utils::globalVariables(".")` to indicate that `.` is a known global variable.
3. You can configure `closed_curly_linter(allow_single_line = TRUE)` to allow `{print(summary(.))}` in a single line.

**Comment by iago-pssjd** [comment]
Thank you @AshesITR .

For the point 1, I assume it is better practice to use `assign` that `<<-`, so I will do that.
Relative to -mainly- 2, but also 3, just a question, can I configure that options in the configuration `.lintr` file in some way?

Thank you again

**Comment by AshesITR** [comment]
Regarding 3, exactly. For example your `.lintr` could read

```
linters: with_defaults(closed_curly_linter(allow_single_line = TRUE))
```

See also the vignette on configuring lintr.

**Comment by MichaelChirico** [comment]
It looks like the issue here is solved; please re-open if that's not the case.
